app/src/org/commcare/network/CTInterceptorConfig.kt

```diff
@@ -0,0 +1,48 @@
+package org.commcare.network
+
+import com.appmattus.certificatetransparency.CTLogger
+import com.appmattus.certificatetransparency.VerificationResult
+import com.appmattus.certificatetransparency.certificateTransparencyInterceptor
+
+import okhttp3.OkHttpClient
+import org.commcare.core.network.HttpBuilderConfig
+import org.commcare.preferences.HiddenPreferences
+import org.commcare.util.LogTypes
+import org.javarosa.core.services.Logger
+
+/**
+ * Adds and removes Certificate Transparency (CT) network interceptor
+ *
+ */
+class CTInterceptorConfig:HttpBuilderConfig  {
+
+
+    override fun performCustomConfig(client: OkHttpClient.Builder): OkHttpClient.Builder {
+        if(HiddenPreferences.isCertificateTransparencyEnabled()) {
+            return client.addNetworkInterceptor(getInterceptor)
+        }
+
+        // In case there are CT Interceptors already attached
+        removeCTInterceptors(client)
+        return client
+    }
+
+    private val getInterceptor = certificateTransparencyInterceptor {
```

Comment:
maybe lazy init this to make sure we are not initialising the interceptor when the certificate transparency is disabled.